docs/spec_draft.md

```diff
@@ -1917,6 +1918,70 @@ and produces a `result` tensor. More formally,
 ```
 
 [Back to Ops](#index-of-ops)
+
+## stablehlo.rng_bit_generator
+
+### Semantics
+
+Returns an `output` filled with uniform random bits and an updated output state
+`output_state` given an initial state `initial_state` using the pseudorandom
+number generator algorithm `rng_algorithm`. The output is guaranteed to be
+deterministic function of `initial_state`, but it is not guaranteed to be
+deterministic between backends and different compiler versions.
+
+`rng_algorithm` is one of the following, different variants of the algorithm is
+implementation-defined:
+  * `DEFAULT`: Backend specific algorithm.
+  * `THREE_FRY`: Threefry counter-based PRNG algorithm.*
+  * `PHILOX`: Philox algorithm to generate random numbers in parallel.*
```

Comment:
2. From what I could tell by the wording in the comment, DEFAULT can be _any_ backend specific prng algorithm, but I'm not 100% sure since common hardwares might already be implementing some variant of Threefry or Philox as the default/common practice or something else entirely:
https://github.com/tensorflow/tensorflow/blob/c82413e988b1dfe12b2ae9c668cd75f3de3fd778/tensorflow/compiler/xla/xla_data.proto#L652-L657
4. I believe Threefry and Philox are both general "algorithms" that work just like different variants of e.g. AES-256 and AES-512 (Here, the "algorithm" is AES). In Threefry, it refers to many "variants" of the "Threefry" algorithm using how many inputs, different number of bits, and # of rounds to compare speed across different hardwares. XLA decided to use a variant Threefry-2-32-20, which uses 2 32-bit input with 20 rounds of shifting, that is not mentioned in the paper. Hence, I'm more inclined to believe that the variant chosen is implementation-defined and `initial_state` follows from the implementer's choice of the variant.

Just fyi, Philox variant used in XLA is 4-32-10. Bit truncation/concat is done to generate other length bits.